lib/semantics/expression.cc

```diff
@@ -1226,17 +1227,17 @@ evaluate::MaybeExpr AnalyzeExpr(evaluate::FoldingContext &context,
 class Mutator {
 public:
   Mutator(evaluate::FoldingContext &context,
-      const IntrinsicTypeDefaultKinds &defaults,
-      const evaluate::IntrinsicProcTable &intrinsics)
-    : context_{context}, defaults_{defaults}, intrinsics_{intrinsics} {}
+      const SemanticsContext &semanticsContext)
+    : context_{context}, semanticsContext_{semanticsContext} {}
 
   template<typename A> bool Pre(A &) { return true /* visit children */; }
   template<typename A> void Post(A &) {}
 
   bool Pre(parser::Expr &expr) {
     if (expr.typedExpr.get() == nullptr) {
       if (MaybeExpr checked{
-              AnalyzeExpr(context_, defaults_, intrinsics_, expr)}) {
+              AnalyzeExpr(context_, semanticsContext_.defaultKinds(),
```

Comment:
Or you could change AnalyzeExpr to just take a SemanticsContext reference and unpack it in there as necessary.  This would make it easier to add things to the semantics context later.